testplan/common/utils/match.py

```diff
@@ -83,15 +290,13 @@ def seek(self, mark: Optional[str] = None):
         :param mark: Name of the mark.
         """
         if mark is None:
-            self.position = 0
+            self.position = self.strategy.seek()
         else:
-            self.position = self.marks[mark]
+            self.position = self.strategy.seek(mark=self.marks[mark])
```

Comment:
here the position you get from the mark should be good, no need to involve the strategy.